tasks/main.yml

```diff
@@ -23,14 +23,26 @@
 
 # The user is created by installing pacemaker packages. We just need to set the
 # password.
-- name: Set hacluster password
-  user:
-    name: hacluster
-    password: "{{
-      ha_cluster_hacluster_password | string |
-      password_hash('sha512', ansible_hostname.replace('-', 'x')[:16]) }}"
+- name: Provide a password for the hacluster user
   when:
     - ha_cluster_hacluster_password | string | length > 0
+  block:
+    - name: Generate a password hash
+      # The arg `-6` means SHA512 based algorithms.
+      command:
+        cmd: >-
+          openssl passwd
+            -6
+            -salt {{ ansible_hostname.replace('-', 'x') }}
+            {{ ha_cluster_hacluster_password | string }}
+      register: __ha_cluster_openssl_call_result
+      changed_when: false
+      no_log: true
+
+    - name: Set hacluster password
+      user:
+        name: hacluster
+        password: "{{ __ha_cluster_openssl_call_result.stdout }}"
```

Comment:
```suggestion
        password: "{{ __ha_cluster_openssl_call_result.stdout }}"
      no_log: true
```

Reply:
I can add no_log, but is it necessary? The password is not shown, the user module hides it even without no_log:
```
TASK [linux-system-roles.ha_cluster : Set hacluster password] ******************
changed: [localhost] => {
    "append": false,
    "changed": true,
    "comment": "cluster user",
    "group": 189,
    "home": "/var/lib/pacemaker",
    "move_home": false,
    "name": "hacluster",
    "password": "NOT_LOGGING_PASSWORD",
    "shell": "/sbin/nologin",
    "state": "present",
    "uid": 189
}
```